src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RPC/SerializableWorkspaceEdit.ts

```diff
@@ -0,0 +1,46 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { SerializableCreateDocument } from './SerializableCreateDocument';
+import { SerializableDeleteDocument } from './SerializableDeleteDocument';
+import { convertRangeFromSerializable } from './SerializableRange';
+import { SerializableRenameDocument } from './SerializableRenameDocument';
+import { SerializableTextEdit } from './SerializableTextEdit';
+
+type SerializableDocumentChange = SerializableCreateDocument | SerializableRenameDocument | SerializableDeleteDocument;
+
+export interface SerializableWorkspaceEdit {
+    changes?: {[key: string]: Array<SerializableTextEdit>};
+    documentChanges?: Array<SerializableDocumentChange>;
+}
+
+export function convertWorkspaceEditFromSerializable(data: SerializableWorkspaceEdit): vscode.WorkspaceEdit {
+    const workspaceEdit = new vscode.WorkspaceEdit();
+    if (Array.isArray(data.documentChanges)) {
+        for (const documentChange of data.documentChanges) {
+            if (documentChange.kind === 'create') {
+                workspaceEdit.createFile(vscode.Uri.parse(documentChange.uri), documentChange.options);
+            } else if (documentChange.kind === 'rename') {
+                workspaceEdit.renameFile(vscode.Uri.parse(documentChange.oldUri), vscode.Uri.parse(documentChange.newUri), documentChange.options);
+            } else if (documentChange.kind === 'delete') {
```

Comment:
What happens if the documentChange is something other than these 3?

Reply:
That's it according to the spec!